public/app/plugins/panel/logs/LogsPanel.tsx

```diff
@@ -151,6 +208,13 @@ export const LogsPanel = ({
     [data]
   );
 
+  /**
+   * Scrolls the given row into view.
+   */
+  const scrollIntoView = (row: HTMLElement) => {
+    row.scrollIntoView(true);
+  };
```

Comment:
```suggestion
  const scrollIntoView = useCallback(row: HTMLElement) => {
    row.scrollIntoView(true);
  }, []);
```

Reply:
Why would we memoize this function?

Reply:
Is there any reason not to? You would be passing a new reference to `LogRows` when nothing changed.

Reply:
I'm confused, this function only takes an `HTMLElement` and doesn't reference anything in the react component?

Reply:
Ahhh, nevermind, still early. I think the overhead of useEffect would be more then recreating this simple function, but we can test it!

Reply:
https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Usually yes, but this is passed to `LogRows`, so it should always be less expensive than a potential cascade of unwanted rerenders.

Reply:
Demo: first browser, without useCallback. Second browser, with useCallback. I only added a `console.log` in `LogRows`.

https://github.com/grafana/grafana/assets/1069378/943251c8-a2c4-41d6-9b5a-9d34e9213ca3

![imagen](https://github.com/grafana/grafana/assets/1069378/4969ead9-c17c-4089-b9ab-0e879b3d98b7)

![imagen](https://github.com/grafana/grafana/assets/1069378/70fb8d2b-1a69-498f-a928-5d5dbb47d2fc)

Reply:
Nice work Matias, I missed that since this function was being passed as a prop it would cause a re-render on resize. Sorry I was being obtuse, great catch and thank you!

Reply:
No worries! Not long ago I spent some time working on this on https://github.com/grafana/grafana/issues/60687 so I keep all these issues and side effects on the back of my head.